meta(changelog): Update changelog for 8.0.0-rc.0 (#11916)
getsentry/sentry-javascript: CHANGELOG.md

- "You miss 100 percent of the chances you don't take. — Wayne Gretzky" — Michael Scott

## 8.0.0-rc.0

This is the first release candidate of Sentry JavaScript SDK v8.

> We also recommend, reading the detailed
> [migration guide](https://docs.sentry.io/platforms/javascript/migration/v7-to-v8/) in the docs.

Reviewer: ```suggestion
We recommend to read the detailed
[migration guide](https://docs.sentry.io/platforms/javascript/migration/v7-to-v8/) in the docs.
```

### Version Support

The Sentry JavaScript SDK v8 now supports Node.js 14.8.0 or higher. This applies to @sentry/node and all of our
node-based server-side sdks (@sentry/nextjs, @sentry/remix, etc.).

The browser SDKs now require ES2018+ compatible browsers. New minimum browser versions:

Chrome 63 Edge 79 Safari/iOS Safari 12 Firefox 58 Opera 50 Samsung Internet 8.2

For more details, please see the

> [version support section in migration guide](https://github.com/getsentry/sentry-javascript/blob/8.0.0-beta.1/MIGRATION.md#1-version-support-changes).

Reviewer: ```suggestion
[version support section in migration guide](https://github.com/getsentry/sentry-javascript/blob/develop/MIGRATION.md#1-version-support-changes).
```

should this be the docs on develop? 🤔

### Important Changes

- **feat(svelte): Add Svelte 5 support (#11807)**:

We now officially support Svelte 5.

- **feat(node): Support Node 22 (#11871)**:

We now officially support Node 22.

> - **fix(node): Fix nest.js error handler (#11874)**

Reviewer: I would move all these fixes down to "Other Changes" - important here is really the big, breaking stuff only!

Author: I added all important changes from the last beta releases plus Node 22 support and the one with Nest.js as the Nest.js fix is a breaking change, right?

The default limit of queued events to be sent was increased from 30 to 64 events. You may observe a higher memory
footprint of the SDK. You can override this limit by setting the `transportOptions.bufferSize` option in
`Sentry.init()`.

- **feat(replay): Add "maxCanvasSize" option for replay canvases (#11617)**:

A `maxCanvasSize` option was added to the `replayCanvasIntegration` to disallow capturing of canvases larger than a
certain size. This value defaults to `1280` which will not capture canvases bigger than 1280x1280 pixels.

- **feat(opentelemetry): Add `addOpenTelemetryInstrumentation` (#11667)**

A utility function `addOpenTelemetryInstrumentation` was added that allows for the registration of instrumentations that
conform to the OpenTelemetry JS API without having to specify `@opentelemetry/instrumentation` as a dependency.

- **ref(core): Don't start transaction for trpc middleware (#11697)**

Going forward, the Sentry `trpcMiddleware` will only create spans. Previously it used to always create a transaction.
This change was made to integrate more nicely with the HTTP instrumentation added in earlier versions to avoid creating
unnecessary transactions.

> - **feat(browser): Update `propagationContext` on `spanEnd` to keep trace consistent**

Reviewer: really all of these should probably be under "other changes" I believe, coming here you have no context of all the other hundreds of changes in the rc 😅

- "You miss 100 percent of the chances you don't take. — Wayne Gretzky" — Michael Scott

## 8.0.0-rc.0

This is the first release candidate of Sentry JavaScript SDK v8. Except for missing ESM support, this version is
considered stable.

We recommend to read the detailed [migration guide](https://docs.sentry.io/platforms/javascript/migration/v7-to-v8/) in
the docs.

> ### Version Support
> 
> The Sentry JavaScript SDK v8 now supports Node.js 14.8.0 or higher. This applies to @sentry/node and all of our
> node-based server-side sdks (@sentry/nextjs, @sentry/remix, etc.).
> 
> The browser SDKs now require ES2018+ compatible browsers. New minimum browser versions:
> 
> Chrome 63 Edge 79 Safari/iOS Safari 12 Firefox 58 Opera 50 Samsung Internet 8.2
> 
> For more details, please see the
> [version support section in migration guide](https://github.com/getsentry/sentry-javascript/blob/develop/MIGRATION.md#1-version-support-changes).
> 

Reviewer: ```suggestion
```

I think we can omit this as well here then, keep it a "normal"/"small" changelog!

### Important Changes

- **feat(node): Support hapi v21 & fix E2E test (#11906)**

We now support hapi v21 and added tests for it.

- **feat(node): Warn if ESM mode is detected (#11914)**

> When running Sentry in ESM mode, you will get a warning about that. Consider transpiling your application to CJS
> (CommonJS) in your production build.

Reviewer: ```suggestion
When running Sentry in ESM mode, we will now warn you that this is not supported as of now. We are working on ensuring support with ESM builds.
```

maybe something like this? 🤔

### Other Changes

> - dev: Upgrade to `jsdom@21.1.2` (#11900)
> - docs: fix typo in doc for `reactRouterV6BrowserTracingIntegration` (#11896)

Reviewer: ```suggestion
```

no need to highlight docs/dev changes in here :)
